### DISCUSSION TITLE
Idea for making format patch more intuitive for single-revision patches

### DISCUSSION THREAD

**Discussion by matta** [magit/magit#4734] [Ideas]
Currently format patch (W c) behaves differently when a range of revisions is marked vs. when a single revision is marked.  I.e. marking a single revision With C-SPC before W c behaves as if no revision is marked.  I suggest changing this such that it is possible to mark a single revision with C-SPC, then press W c, and create a patch with just that revision.  Magit does allow me to manually type out that single revision to create a single revision patch.

I think this will bring things in line with other things in Emacs that do something with the active region when it is active.  I can't think of anything in Emacs that ignores the active region if it doesn't satisfy a specific "shape" (the command might issue an error in that case instead).

**Comment by yurikhan** [comment]
Magit generally doesn’t use the region to identify a single commit; rather, it usually takes the commit hash, branch name, or tag name at or near point. For some reason, `magit-patch-create` does not go through the machinery that identifies those. I suggest that it should.

**Reply by tarsius** [reply]
> Magit generally doesn’t use the region to identify a single commit; rather, it usually takes the commit hash, branch name, or tag name at or near point.

That used to be the case but especially in cases like this, where the alternative is to prompt instead, selecting one section now usually acts on that section. But there are probably other places where that change hasn't been made yet.

**Comment by tarsius** [comment]
> I suggest changing this such that it is possible to mark a single revision with C-SPC, then press W c, and create a patch with just that revision.

I agree.

> Magit does allow me to manually type out that single revision to create a single revision patch.

The commit at point is actually the default. The default isn't displayed because of complications caused by the use of `completing-read-multiple`. I would have to look into that in more detail to figure out how I want to improve that.

I am on a break, so I won't do that soon. Meanwhile you can just type `RET` at the prompt.

**Reply by tarsius** [reply]
> I suggest changing this such that it is possible to mark a single revision with C-SPC, then press W c, and create a patch with just that revision.

I finally got around to that, in 47e0c3c2e9f0eb2934a3659a0ef64097e50805b1.
